### DISCUSSION TITLE
Mistimed South Pacific swells - perhaps a time step issue?

### DISCUSSION THREAD

**Discussion by MicahDavid** [NOAA-EMC/WW3#1282] [General]
Hi,

I am running an operational 0.5 degree global grid, and I'm noticing that South Pacific swells are arriving at the US coastline faster than what is seen on buoy data.  Around 12-24 hours early.

I'm forcing with GFS winds, using the ST4 switch, and changing BETAMAX
to 1.33 (for GFS winds).

My first guess of the possible problem is that the model time steps are too large. So far my attempts at lowering the time steps have not made a discernible difference.

Original timesteps:
3600. 480 1800 30.

The changed timesteps that didn't make much difference are:
1800. 600. 900. 30.

My next attempt is to test at:
1200. 600. 600. 30.

Is it reasonable that the time steps could be the source of the timing problem? I've double checked the wind and ice input, and I believe these inputs are good. Are there any other possible sources that I could investigate as a source of the problem?

Here's the grid input file:

  'Global 30 min wave grid       '
$
   1.10  0.035  29  24  0.5
$
   F T T T F T
$
  1800.  600. 900.  30.
$
  &MISC CICE0 = 0.25, CICEN = 0.75, FLAGTR = 4 /
  &SIN4 BETAMAX = 1.33, /
END OF NAMELISTS
$
    'RECT'  T  'SMPL'
    720    311
      0.50   0.50   1.
      0.   -77.5    1.
     -0.1 2.50  20  0.001 1 1 '(....)' 'NAME' 'glo_30m.bot'
                21  0.01  1 1 '(....)' 'NAME' 'glo_30m.obst'
                22        1 1 '(....)' 'NAME' 'glo_30m.mask'

Here are my switches: 
F90 NCEP2 MPI DIST PR3 UQ ST4 FLX0 STAB0 LN1 NL1 BT1 DB1 TR0 BS0 IC0 IS0 REF0 XX0 WNT1 WNX1 CRT1 CRX1 O0 O1 O2 O3 O4 O5 O6 O7 O10 O11 MLIM

Thank you for any suggestions.

**Comment by mickaelaccensi** [comment]
Hi,

You should try with smaller time steps.

The first time step to calculate is the maximum CFL time step which depends on the lowest frequency Fmin previously set to 0.035Hz and the lowest spatial grid resolution along x in meter at the latitude closest to the pole, here 77.5S.
```
dx = min(reslon * cos(maxlat*Pi/180)*1852*60, reslon * cos(minlat*Pi/180)*1852*60)
dx = 0.5 * cos(-77.5*3.14/180)*1852*60 = 12062m
```
	with reslon the resolution along x in degrees
	with maxlat the northern latitude in degrees
	with minlat the southern latitude in degrees
reminder : 1 degree=60minutes // 1minute=1mile // 1mile=1.852km

The formula for the CFL time is :
```
Tcfl = dx / (g / (Fmin*4*Pi) ) with the constants Pi=3,14 and g=9.8m/s²;
Tcfl = 12062 / (9.8 / (0.035*4*3.14) ) = 541s
maxTcfl ~= 90% Tcfl = 487s => 480s
```
set it to 480s to be multiple of 60s

Knowing the maxTcfl, the maximum global time step is usually set to 3 times bigger, it can be set up to 6 times bigger to save computation time despite a loss of precision :
`Tglob = 3*maxTcfl = 3*480 = 1440s`
The refraction time step depends on how strong can be the wind or current on your grid, it is usually set to 6 times less the global time step but it can be decreased to 10 times less in case of strong wind or current:
`Tref = Tglob / 6	 = 1440 / 6 = 240s`
The minimum source term time step is usually defined between 1s and 10s.
`Tsrc = 1s`

so you should use : 

`1440 480 240 1`
